src/transformState.js

```diff
@@ -3,9 +3,39 @@
 /**
  * @param {Object} state
  * @param {Object[]} actions
- */
+*/
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeSpecificKeys(action);
+        break;
+
+      case 'clear':
+        removeAllKeys();
+        break;
+
+      default:
+        throw new Error(`No ${action.type} action type found`);
+    }
+  }
+
+  function removeAllKeys() {
```

Comment:
Add the 'state' parameter to the 'removeAllKeys' function, so it can access and modify the state object.